From a8e89facf49ba2f6aa2b225fdb3ef349c574de90 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 11 Aug 2015 15:52:10 +0800
Subject: [PATCH 0334/1691] MLK-11340-43 usb: chipidea: imx: fix usb charger
 detection init NULL pointer

commit a8e89facf49ba2f6aa2b225fdb3ef349c574de90 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Dereference the charger->psy after the power_supply is created to fix the
null pointer issue.

Signed-off-by: Li Jun <b47624@freescale.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index e969156..e1e4040 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -396,7 +396,6 @@ static int imx_usb_charger_get_property(struct power_supply *psy,
 static int imx_usb_register_charger(struct usb_charger *charger,
 		const char *name)
 {
-	struct power_supply		*psy = charger->psy;
 	struct power_supply_desc	*desc = &charger->psy_desc;
 
 	if (!charger->dev)
@@ -414,15 +413,16 @@ static int imx_usb_register_charger(struct usb_charger *charger,
 	desc->properties	= imx_usb_charger_power_props;
 	desc->num_properties	= ARRAY_SIZE(imx_usb_charger_power_props);
 	desc->get_property	= imx_usb_charger_get_property;
-	psy->supplied_to	= imx_usb_charger_supplied_to;
-	psy->num_supplicants	= sizeof(imx_usb_charger_supplied_to)
-					/ sizeof(char *);
 
-	charger->psy = devm_power_supply_register(charger->dev->parent,
+	charger->psy = devm_power_supply_register(charger->dev,
 						&charger->psy_desc, NULL);
 	if (IS_ERR(charger->psy))
 		return PTR_ERR(charger->psy);
 
+	charger->psy->supplied_to	= imx_usb_charger_supplied_to;
+	charger->psy->num_supplicants	= sizeof(imx_usb_charger_supplied_to)
+					/ sizeof(char *);
+
 	return 0;
 }
 
-- 
1.9.1

